src/wallet/wallet.cpp

```diff
@@ -1423,6 +1427,27 @@ void CWallet::transactionAddedToMempool(const CTransactionRef& tx) {
     if (it != mapWallet.end()) {
         RefreshMempoolStatus(it->second, chain());
     }
+
+    const Txid& conflict_txid = tx->GetHash();
+
+    for (const CTxIn& tx_in : tx->vin) {
+        auto range =  mapTxSpends.equal_range(tx_in.prevout);
+
+        // For each wallet transaction spending this prevout...
+        for (auto _it = range.first; _it != range.second; ++_it) {
+            // if this tx is the same as the tx which was just added to the mempool, continue
+            CWalletTx& wallet_tx = mapWallet.at(_it->second);
+            if (wallet_tx.tx->GetHash() == conflict_txid) continue;
+
+            auto try_updating_state = [&](CWalletTx& wtx) EXCLUSIVE_LOCKS_REQUIRED(cs_wallet) {
+                wtx.mempool_conflicts.insert(conflict_txid);
+
+                return TxUpdate::CHANGED;
```

Comment:
In commit "wallet: track mempool conflicts" (a650caab9f11be7f5927f9aa556bb7d2a8ebed33)

Would be nice to add a check that conflict txid is actually inserted as assumed:

```diff
-                wtx.mempool_conflicts.insert(conflict_txid);
-
+                Assume(wtx.mempool_conflicts.insert(conflict_txid).second);
```

Reply:
It seems like that is not always true, so I have changed this code to do something similar to `transactionRemovedFromMempool` where it only returns `TxUpdate::CHANGED` if the txid is actually inserted.